### PR TITLE
Add OStreamLogRecordExporterFactory

### LINKED ISSUE #2239: [OSTREAM EXPORTER] Add factory for `OStreamLogRecordExporter`
**Describe the solution you'd like**
Just like `OStreamMetricExporterFactory` and `OStreamSpanExporterFactory` . Add a factory for OStreamLogRecordExporter`.

### PR DESCRIPTION
**Pull request by owent**
Fixes #2239 

## Changes

+ Add `OStreamLogRecordExporterFactory`

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

File: exporters/ostream/CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,7 @@ if(OPENTELEMETRY_INSTALL)
     DIRECTORY include/opentelemetry/exporters/ostream
     DESTINATION include/opentelemetry/exporters
     PATTERN "*.h"
-    PATTERN "log_Exporter.h" EXCLUDE)
+    PATTERN "log_record_exporter.h" EXCLUDE)
```
Comment (lalitb) [inline exporters/ostream/CMakeLists.txt:30]: I can't recollect why this exclude was added, do we really need this?

**Review by lalitb — APPROVED** [review]
LGTM. Thanks for the PR.

File: exporters/ostream/CMakeLists.txt (hunk 1 above)
Comment (lalitb) [reply exporters/ostream/CMakeLists.txt:30]: Ok. probably copying this separately if WITH_LOGS_PREVIEW is enabled. In that case, should we also exclude `log_record_exporter_factory.h` here?

**Comment by marcalff** [comment]
Open comment marked as resolved, so merging now.

**Pull request merged**